scripts/bug_retriever.py

```diff
@@ -20,161 +20,172 @@ def retrieve_bugs(self, limit: int = None) -> None:
         db.download(bugzilla.BUGS_DB)
 
         # Get IDs of bugs changed since last run.
-        last_modified = db.last_modified(bugzilla.BUGS_DB)
-        logger.info(
-            f"Retrieving IDs of bugs modified since the last run on {last_modified}"
-        )
-        changed_ids = set(
-            bugzilla.get_ids(
-                {"f1": "delta_ts", "o1": "greaterthaneq", "v1": last_modified.date()}
-            )
-        )
-        logger.info(f"Retrieved {len(changed_ids)} IDs.")
-
-        all_components = bugzilla.get_product_component_count(9999)
-
-        deleted_component_ids = set(
-            bug["id"]
-            for bug in bugzilla.get_bugs()
-            if "{}::{}".format(bug["product"], bug["component"]) not in all_components
-        )
-        logger.info(
-            f"{len(deleted_component_ids)} bugs belonging to deleted components"
-        )
-        changed_ids |= deleted_component_ids
-
-        # Get IDs of bugs between (two years and six months ago) and now.
-        two_years_and_six_months_ago = datetime.utcnow() - relativedelta(
-            years=2, months=6
-        )
-        logger.info(f"Retrieving bug IDs since {two_years_and_six_months_ago}")
-        timespan_ids = bugzilla.get_ids_between(two_years_and_six_months_ago)
-        if limit:
-            timespan_ids = timespan_ids[-limit:]
-        logger.info(f"Retrieved {len(timespan_ids)} IDs.")
-
-        # Get IDs of labelled bugs.
-        labelled_bug_ids = labels.get_all_bug_ids()
-        if limit:
-            labelled_bug_ids = labelled_bug_ids[-limit:]
-        logger.info(f"{len(labelled_bug_ids)} labelled bugs to download.")
-
-        # Get the commits DB, as we need it to get the bug IDs linked to recent commits.
-        # XXX: Temporarily avoid downloading the commits DB when a limit is set, to avoid the integration test fail when the commits DB is bumped.
-        if limit is None:
-            assert db.download(repository.COMMITS_DB)
-
-        # Get IDs of bugs linked to commits (used for some commit-based models, e.g. backout and regressor).
-        start_date = datetime.now() - relativedelta(years=3)
-        commit_bug_ids = list(
-            set(
-                commit["bug_id"]
-                for commit in repository.get_commits()
-                if commit["bug_id"]
-                and dateutil.parser.parse(commit["pushdate"]) >= start_date
+        try:
+            last_modified = db.last_modified(bugzilla.BUGS_DB)
+        except Exception as e:
+            if str(e) == "Last-Modified is not available":
```

Comment:
When the exception is thrown, last_modified is undefined and the next `if last_modified` will fail.
You should do the same as github_issue_retriever (define `last_modified` as None first), or assign a value to it here

Reply:
Ok

Reply:
Done